internal/provider/jira_product_configuration_resource.go

```diff
@@ -0,0 +1,524 @@
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					boolDefault(false),
+				},
+			},
+
+			"risk_acceptance_expiration_notification": {
+				MarkdownDescription: "Send Risk Acceptance expiration notifications as comments",
+				Optional:            true,
+				Type:                types.BoolType,
+				Computed:            true,
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					boolDefault(false),
+				},
+			},
+
+			"jira_instance_id": {
+				MarkdownDescription: "The ID of the Jira Instance to use for this Product",
+				Optional:            true,
+				Type:                types.StringType,
+			},
+
+			"product_id": {
+				MarkdownDescription: "The ID of the Product to configure",
+				Required:            true,
+				Type:                types.StringType,
+			},
+
+			"engagement_id": {
+				MarkdownDescription: "The ID of the Engagement. Can be empty",
+				Optional:            true,
+				Type:                types.StringType,
+			},
+
+			"id": {
+				Computed:            true,
+				MarkdownDescription: "Identifier",
+				PlanModifiers: tfsdk.AttributePlanModifiers{
+					tfsdk.UseStateForUnknown(),
+				},
+				Type: types.StringType, // the id (for import purposes) MUST be a string
+			},
+		},
+	}, nil
+}
+
+func (t jiraProductConfigurationResourceType) NewResource(ctx context.Context, in tfsdk.Provider) (tfsdk.Resource, diag.Diagnostics) {
+	provider, diags := convertProviderType(in)
+
+	return jiraProductConfigurationResource{
+		provider: provider,
+	}, diags
+}
+
+type jiraProductConfigurationResourceData struct {
+	ProjectKey                           types.String `tfsdk:"project_key"`
+	IssueTemplateDir                     types.String `tfsdk:"issue_template_dir"`
+	PushAllIssues                        types.Bool   `tfsdk:"push_all_issues"`
+	EnableEngagementEpicMapping          types.Bool   `tfsdk:"enable_engagement_epic_mapping"`
+	PushNotes                            types.Bool   `tfsdk:"push_notes"`
+	ProductJiraSlaNotification           types.Bool   `tfsdk:"product_jira_sla_notification"`
+	RiskAcceptanceExpirationNotification types.Bool   `tfsdk:"risk_acceptance_expiration_notification"`
+	JiraInstance                         types.String `tfsdk:"jira_instance_id"`
+	Product                              types.String `tfsdk:"product_id"`
+	Engagement                           types.String `tfsdk:"engagement_id"`
+	Id                                   types.String `tfsdk:"id"`
+}
+
+type jiraProductConfigurationResource struct {
+	provider provider
+}
+
+func (r jiraProductConfigurationResource) Create(ctx context.Context, req tfsdk.CreateResourceRequest, resp *tfsdk.CreateResourceResponse) {
+	var data jiraProductConfigurationResourceData
+
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var productIdNumber, engagementIdNumber, jiraInstanceIdNumber int
+	var err error
+	if !data.Product.IsNull() {
```

Comment:
Shouldn't be the case but Product ID and Jira Instance ID are required fields I think - would you go past this if the product doesn't exist?

Reply:
Checked via curl and got this response: 

``` JSON
{
  "non_field_errors":["Either engagement or product has to be set."],
  "message":"{'non_field_errors': [ErrorDetail(string='Either engagement or product has to be set.', code='invalid')]}"
}
```

Since the attributes requires product id. I think this is a non-issue

Reply:
Yeah I think what will happen is it will skip this and then the create call below would fail. But strangely the OpenAPI spec doesn't say that these ids are required, and I don't think the api call actually fails. I don't know exactly why but I think it allows you to create a completely empty jira_product_configuration with no IDs of anything set.

Reply:
> Checked via curl and got this response:

Oh sorry, I didn't see this second comment before I replied. Gotcha... so actually I should mark them both as "optional" fields technically. I will look and see if there's a way to do this kind of validation where one of two fields is required.

Reply:
Turns out to not be that hard, thanks for the poke on this: https://github.com/doximity/terraform-provider-defectdojo/pull/37/commits/1fca77aa0ddfe6966e9dc4335012d02291ad5323